ACPs/113-provable-randomness/README.md

```diff
@@ -0,0 +1,165 @@
+
+The changes proposed here would affect the way a block is being validated. Therefore, when this change gets implemented, it needs to be deployed as a mandatory upgrade.
+
+```
+		+-----------------------+           +-----------------------+
+		|  Block n              | <-------- |  Block n+1            |
+		+-----------------------+           +-----------------------+
+		|  VRF-Sig(n)           |           |  VRF-Sig(n+1)         |
+		|  ...                  |           |  ...                  |
+		+-----------------------+           +-----------------------+
+
+		+-----------------------+           +-----------------------+
+		|  VM n                 |           |  VM n+1               |
+		+-----------------------+           +-----------------------+
+		|  VRF-Out(n)           |           |  VRF-Out(n+1)         |
+		+-----------------------+           +-----------------------+
+
+		VRF-Sig(n+1) = Sign(VRF-Sig(n), Block n+1 proposer's BLS key)
+		VRF-Out(n) = Hash(VRF-Sig(n))
+```
+
+### Changes Details
+
+#### Step 1. Adding BLS signature to proposed blocks
+
+```golang
+type statelessUnsignedBlock struct {
+	…
+	vrfSig    []byte `serialize:”true”`
+}
+```
+
+#### Step 2. Populate signature
+
+When a block proposer attempts to build a new block, it would need to use the parent block as a reference.
+
+The `vrfSig` field within each block is going to be daisy-chained to the `vrfSig` field from it's parent block.
+
+Populating the `vrfSig` would following this logic:
+
+1. The current proposer has a BLS key
+   
+	a. If the parent block has an empty `vrfSig` signature, that signature would be signed using the proposer’s BLS key. This is the base case.
+
+	b. If the parent block does not have an empty `vrfSig` signature, the proposer would sign the bootStrappingBlockSignature with its BLS key. See the bootStrappingBlockSignature details below.
+
+2. The current proposer does not have a BLS key
+   
+   a. If the parent block has a non empty `vrfSig` signature, the proposer would set the proposed block `vrfSig` to the 32 byte hash result of the following preimage:
+	```
+	+-------------------------+----------+------------+
+	|  prefix :               | [8]byte  | "rng-derv" |
+	+-------------------------+----------+------------+
+	|  vrfSig :               | [96]byte |  96 bytes  |
+	+-------------------------+----------+------------+
+	```
+
+	b. If the parent block has an empty `vrfSig` signature, the proposer would leave the field empty.
+
+The bootStrappingBlockSignature that would be used above is the hash of the following preimage:
+
+```
++-----------------------+----------+------------+
+|  prefix :             | [8]byte  | "rng-root" |
++-----------------------+----------+------------+
+|  chainID :            | [32]byte |  32 bytes  |
++-----------------------+----------+------------+
+|  networkID:           | uint32   |  4 bytes   |
```

Comment:
Small nit: It may make more sense to put `networkID` before `chainID` here. Existing serializations (for Warp messages, P-Chain txs) that have those fields put them in that order to go from more general to more specific.

Reply:
makes sense.

Reply:
done